### PR TITLE
uri_template: verbose errors

### PR DESCRIPTION
**Pull request by adisuissa**
Commit Message: uri_template: verbose errors
Additional Description:
Adds verbose error messages explaining why a config is invalid.

Risk Level: low
Testing: Updated tests.
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by adisuissa** [comment]
Related to #34507.
Assigning @yanjunxiang-google for a code-owner first pass.
/assign @yanjunxiang-google

**Comment by tyxia** [comment]
@yanjunxiang-google Friendly ping (as maintainer on-call responsibility) :) 

Once it is LGTY, I will pass it onto Alyssa

File: source/extensions/path/uri_template_lib/uri_template_internal.cc [hunk 1]
```diff
@@ -155,24 +157,25 @@ absl::StatusOr<ParsedResult<Operator>> parseOperator(absl::string_view pattern)
...
   }
-  return absl::InvalidArgumentError("Invalid Operator");
+  return absl::InvalidArgumentError(fmt::format("Invalid Operator: \"{}\"", pattern));
 }
 
 absl::StatusOr<ParsedResult<Variable>> parseVariable(absl::string_view pattern) {
   // Locate the variable pattern to parse.
   if (pattern.size() < 2 || (pattern)[0] != '{') {
-    return absl::InvalidArgumentError("Invalid variable");
+    return absl::InvalidArgumentError(fmt::format("Invalid variable: \"{}\"", pattern));
```
Comment (yanjunxiang-google) [inline source/extensions/path/uri_template_lib/uri_template_internal.cc:166]: Do we have test for this error?

File: source/extensions/path/uri_template_lib/uri_template_internal.cc [hunk 2]
```diff
@@ -155,24 +157,25 @@ absl::StatusOr<ParsedResult<Operator>> parseOperator(absl::string_view pattern)
   if (absl::StartsWith(pattern, "*")) {
     return ParsedResult<Operator>(Operator::PathGlob, pattern.substr(1));
   }
-  return absl::InvalidArgumentError("Invalid Operator");
+  return absl::InvalidArgumentError(fmt::format("Invalid Operator: \"{}\"", pattern));
```
Comment (yanjunxiang-google) [inline source/extensions/path/uri_template_lib/uri_template_internal.cc:160]: I am trying to see whether we have a test for each added error message.

Do we have test for this error?

File: source/extensions/path/uri_template_lib/uri_template_internal.cc [hunk 3]
```diff
@@ -204,7 +207,8 @@ absl::StatusOr<ParsedResult<Variable>> parseVariable(absl::string_view pattern)
     var.match_.push_back(match);
     if (!pattern_item.empty()) {
       if (pattern_item[0] != '/' || pattern_item.size() == 1) {
-        return absl::InvalidArgumentError("Invalid variable match");
+        return absl::InvalidArgumentError(
+            fmt::format("Invalid variable match: \"{}\"", pattern_item));
```
Comment (yanjunxiang-google) [inline source/extensions/path/uri_template_lib/uri_template_internal.cc:211]: Do we have test for this error?

File: source/extensions/path/uri_template_lib/uri_template_internal.cc [hunk 4]
```diff
@@ -275,7 +283,7 @@ absl::StatusOr<ParsedPathPattern> parsePathPatternSyntax(absl::string_view path)
 
   static const LazyRE2 printable_regex = {"^/[[:graph:]]*$"};
   if (!RE2::FullMatch(path, *printable_regex)) {
-    return absl::InvalidArgumentError("Invalid pattern");
+    return absl::InvalidArgumentError(fmt::format("Invalid pattern: \"{}\"", path));
```
Comment (yanjunxiang-google) [inline source/extensions/path/uri_template_lib/uri_template_internal.cc:286]: Do we have test for this error?

**Comment by yanjunxiang-google** [comment]
LGTM

**Review by yanjunxiang-google — APPROVED** [review]

**Comment by tyxia** [comment]
/assign @alyssawilk 

Code owner and maintainer pass.

**Review by alyssawilk — DISMISSED** [review]
LGTM but needs main merge
/wait

**Pull request merged**